examples/ct/structures_test.go

```diff
@@ -147,6 +149,104 @@ func TestSignV1SCTForPrecertificate(t *testing.T) {
 	assert.Equal(t, cert.RawTBSCertificate, leaf.TimestampedEntry.PrecertEntry.TBSCertificate, "tbs cert mismatch")
 }
 
+func TestSerializeMerkleTreeLeafBadVersion(t *testing.T) {
+	var leaf ct.MerkleTreeLeaf
+
+	leaf.Version = 199 // Out of any expected valid range
+
+	var w io.Writer
+	err := WriteMerkleTreeLeaf(w, leaf)
+	assert.Error(t, err, "incorrectly serialized leaf with bad version")
+}
```

Comment:
So it would be _nice_ to assert that w wasn't written to because of errors ... dependent on my comment above.
I guess it depends how much this is example code versus something more robust.  (or whether you think a partially written-to stream matters)